reps/2023-10-13-accelerator-support.md

```diff
@@ -0,0 +1,399 @@
+# Ray Accelerator Support
+
+## Summary
+
+Holistic design of supporting accelerators in Ray.
+This also serves as the reference doc for contributors
+to add new accelerator support in the future.
+
+### General Motivation
+
+Nowadays, more and more specialized accelerators (e.g. TPU, HPU) are coming out to speed up AI workloads.
+Supporting those accelerators natively (similar to how Ray supports Nvidia GPUs) in Ray will increase Ray's adoption as the computing framework for scaling AI applications.
+
+### Should this change be within `ray` or outside?
+
+Inside `ray` project since we want to support those accelerators out of the box.
+
+## Stewardship
+
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experienced Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@pcmoritz, @ericl, @scv119
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+@scv119
+
+## Design
+
+### Ray Core
+
+#### API
+Users will use accelerators through Ray resources. For GPU accelerators (e.g. Nvidia, AMD, Intel GPUs), they will use the `GPU` resource name. For other accelerators, they will each have their own resource name (e.g. TPU accelerator has resource name `TPU`).
+
+| Resource    | Accelerator               |
+| ----------- | ------------------------- |
+| GPU         | Nvidia, Intel, AMD, Apple |
+| TPU         | TPU                       |
+| HPU         | Habana Gaudi              |
+| neuron_cores| AWS Trainium, Inferentia  |
+| ...         | ...                       |
+
+`accelerator_type` option can also be used to specify a particular type of accelerator for a task or actor.
+
+```python
+@ray.remote(num_gpus=1)
+def task():
+    ...
+
+@ray.remote(num_gpus=1, accelerator_type="A100")
```

Comment:
How would say that you are OK with either A100 or A10G?

Reply:
accelerator_type is just labels/custom resources that Ray adds for you automatically. For your case, you can just manually create your own label/custom resource like `A100_OR_A10G` and request it in your task.

Reply:
Later on, once we switch to label you can say `"accelerator_type": In("A100", "A10G")`

Reply:
Maybe we can allow syntactic sugar like accelerator_type="A100,A10"? Feel like this is more pythonic API

Reply:
there's also the proposed placement group alternatives feature which may be a better way of solving that

Reply:
what was the solution/conclusion that we landed on?